cpp/src/arrow/dataset/partition_test.cc

```diff
@@ -189,6 +189,12 @@ TEST_F(TestPartitioning, Partition) {
                   expected_expressions);
 }
 
+TEST_F(TestPartitioning, DefaultPartitioningIsDirectoryPartitioning) {
+  auto partitioning = Partitioning::Default();
+  ASSERT_EQ(partitioning->type_name(), "directory");
+  AssertSchemaEqual(partitioning->schema(), schema({}));
+}
+
```

Comment:
Can you also add a test to verify directory partitioning works with an empty schema?

```

TEST_F(TestPartitioning, DirectoryPartitioningEmpty) {
  partitioning_ = std::make_shared<DirectoryPartitioning>(schema({}));
  written_schema_ = partitioning_->schema();

  // No partitioning info
  AssertParse("", literal(true));
  // Files can be in subdirectories
  AssertParse("/foo/", literal(true));
  // Partitioning info is discarded on write
  AssertFormat(equal(field_ref("alpha"), literal(7)), "");
}
```

Reply:
Sure!
Done!